Tests/ApolloCodegenTests/ApolloCodegenTests.swift

```diff
@@ -142,6 +142,75 @@ class ApolloCodegenTests: XCTestCase {
     expect(try ApolloCodegen.compileGraphQLResult(config).operations).to(haveCount(2))
   }
 
+  func test_CCN_compileResults_givenOperations_withNoErrors_shouldReturn() throws {
+    let schemaData: Data = {
+      """
+      type Query {
+        author: Author
+      }
+
+      type Author {
+        name: String
+        age: Int
+      }
+      """
+    }().data(using: .utf8)!
+    // given
+    let schemaPath = createFile(containing: schemaData, named: "schema.graphqls")
+
+    let authorsData: Data =
+      """
+      query getAuthors {
+        author! {
+          name!
+        }
+      }
+      """.data(using: .utf8)!
+    createFile(containing: authorsData, named: "authors-operation.graphql")
+
+    let config = ReferenceWrapped(value: ApolloCodegenConfiguration.mock(input: .init(
+      schemaPath: schemaPath,
+      searchPaths: [directoryURL.appendingPathComponent("*.graphql").path]
+    )))
+
+    let compiledDocument = try ApolloCodegen.compileGraphQLResult(
+      config,
+      experimentalFeatures: .init(clientControlledNullability: true)
+    )
+
+    // then
+    expect(compiledDocument.operations).to(haveCount(1))
+  }
+
+  func test_CCN_compileResults_givenOperations_withErrors_shouldError() throws {
+    // given
+    let schemaPath = createFile(containing: schemaData, named: "schema.graphqls")
+
+    let authorsData: Data =
+      """
+      query getAuthors {
+        authors {
+          name!
+        }
+      }
+      """.data(using: .utf8)!
+    createFile(containing: authorsData, named: "authors-operation.graphql")
+
+    let config = ReferenceWrapped(value: ApolloCodegenConfiguration.mock(input: .init(
+      schemaPath: schemaPath,
+      searchPaths: [directoryURL.appendingPathComponent("*.graphql").path]
+    )))
+
+    // then
+    expect(try ApolloCodegen.compileGraphQLResult(config).operations).to(throwError { error in
+      guard let error = error as? GraphQLError else {
+        fail("Expected .graphQLSourceValidationFailure, got .\(error)")
```

Comment:
What about this causes the validation error?

Reply:
In this test we're attempting to compile a document that contains CCN operators without CCN enabled. I've added some more info to the failure message to make that more obvious.

Reply:
Ahhhh thank you!